### PR TITLE
Disable webaudio whitelist and add pls files to blacklisted filetypes

### PR DESCRIPTION
**Pull request by plally**
- Add hook for checking whitelisted URLs
- Add hook to disable stream content checking
- Check for PLS files, PLS files can also be used to bypass the whitelist similar to M3U

### CODE REVIEW

File: lua/autorun/webaudio.lua [hunk 1]
```diff
@@ -681,6 +681,9 @@ local function isWhitelistedURL(url)
 	if not isstring(url) then return false end
 	url = url:Trim()
 
+	local isWhitelisted = hook.Run("WA_IsWhitelistedURL", url)
```
Comment (Vurv78) [inline lua/autorun/webaudio.lua:684]: Inline this to just ``if hook.Run("WA_IsWhitelistedURL", url) then return true end``

**Review by Vurv78 — CHANGES_REQUESTED** [review]
Just a nitpick, looks good other than that, thanks for the security patch as well.

File: lua/autorun/webaudio.lua (hunk 1 above)
Comment (plally) [reply lua/autorun/webaudio.lua:684]: The intention behind doing it with the ~= nil check was to allow returning `true` or `false` from the hook. 
So the addon defining the hook could also return false to explicitly block a url early.

Comment (plally) [reply lua/autorun/webaudio.lua:684]: That being said, I'm fine changing it if that's not something you want.

Comment (Vurv78) [reply lua/autorun/webaudio.lua:684]: I doubt multiple addons would be working with this hook, but you can keep the `~= nil` logic I guess

**Pull request merged**